compiler/rustc_span/src/lib.rs

```diff
@@ -2159,9 +2159,7 @@ where
         };
 
         Hash::hash(&TAG_VALID_SPAN, hasher);
-        // We truncate the stable ID hash and line and column numbers. The chances
-        // of causing a collision this way should be minimal.
-        Hash::hash(&(file.name_hash as u64), hasher);
+        Hash::hash(&file.name_hash, hasher);
```

Comment:
It seems like this actually has a perf effect after all. The benchmarked version was merging the two u64s of the stable hash and hashing the resulting u64.

Reply:
This landed out of order. The PR that makes this not a regression: https://github.com/rust-lang/rust/pull/110410 is still in queue